app/services/exchange_deleter.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class ExchangeDeleter
+  def delete_incoming(child, parent)
+    Exchange.
+      joins(:order_cycle).
+      where(sender: parent, receiver: child, incoming: true).
+      destroy_all
```

Comment:
This is a very small service. It could probably just be on the Permission model.